app/models/piece.rb

```diff
@@ -1,4 +1,39 @@
 class Piece < ActiveRecord::Base
   belongs_to :game
   belongs_to :player
+
+  def obstructing_pieces(x_destination, y_destination)
+    x_start = x_position
+    y_start = y_position
+    @x_destination = x_destination
+    @y_destination = y_destination
+    pieces = []
+
+    if moving_horizontally?
+      pieces = Piece.where(game_id: game.id, y_position: y_start, x_position: (x_start + 1..x_destination - 1)).exists?
```

Comment:
A lot of repetition here, what if you instead
* set default variables for `y_position` (y_start) and `x_position` (x_start) 
* then modify them in the ifs (`_start + 1..x_destination - 1)`)
* then perform one query afterwards on the vars